src/main/java/org/mastodon/mamut/tomancak/LineageRandomColorizer.java

```diff
@@ -0,0 +1,178 @@
+
+	// sets up the drop-down box in the dialog with two create-tagset-items
+	// followed by the list of the currently available tagsets
+	private void readAvailColors() {
+		List<String> choices = new ArrayList<>(50);
+
+		pluginAppModel.getAppModel().getModel()
+				.getTagSetModel()
+				.getTagSetStructure()
+				.getTagSets()
+				.forEach( ts -> choices.add( "Use existing "+ts.getName() ) );
+
+		//finally, also allow to create new tag sets (color palettes)
+		choices.add( COLORS_16 );
+		choices.add( COLORS_64 );
+
+		this.getInfo().getMutableInput( "colorScheme", String.class ).setChoices( choices );
+	}
+
+	static final Map<String, Integer> mokolePaletteOf16Colors = new HashMap<>(16);
+	{
+		// used https://mokole.com/palette.html with the settings:
+		// 16 colors, 5% min luminosity, 90% max luminosity, 5000 max loops
+		mokolePaletteOf16Colors.put( "darkslategray", rgbToValidColor(0x2f4f4f) );
+		mokolePaletteOf16Colors.put( "saddlebrown", rgbToValidColor(0x8b4513) );
+		mokolePaletteOf16Colors.put( "darkgreen", rgbToValidColor(0x006400) );
+		mokolePaletteOf16Colors.put( "darkkhaki", rgbToValidColor(0xbdb76b) );
+		mokolePaletteOf16Colors.put( "navy", rgbToValidColor(0x000080) );
+		mokolePaletteOf16Colors.put( "mediumturquoise", rgbToValidColor(0x48d1cc) );
+		mokolePaletteOf16Colors.put( "red", rgbToValidColor(0xff0000) );
+		mokolePaletteOf16Colors.put( "orange", rgbToValidColor(0xffa500) );
+		mokolePaletteOf16Colors.put( "yellow", rgbToValidColor(0xffff00) );
+		mokolePaletteOf16Colors.put( "lime", rgbToValidColor(0x00ff00) );
+		mokolePaletteOf16Colors.put( "mediumspringgreen", rgbToValidColor(0x00fa9a) );
+		mokolePaletteOf16Colors.put( "blue", rgbToValidColor(0x0000ff) );
+		mokolePaletteOf16Colors.put( "orchid", rgbToValidColor(0xda70d6) );
+		mokolePaletteOf16Colors.put( "fuchsia", rgbToValidColor(0xff00ff) );
+		mokolePaletteOf16Colors.put( "dodgerblue", rgbToValidColor(0x1e90ff) );
+		mokolePaletteOf16Colors.put( "lightpink", rgbToValidColor(0xffb6c1) );
+	}
+
+	private TagSetStructure.TagSet createCoolSmallTagSet() {
+		return addNewTagSetToModel( pluginAppModel.getAppModel().getModel(),
+				"Palette of 16 colors", mokolePaletteOf16Colors.entrySet() );
+	}
+
+
+	private TagSetStructure.TagSet createNewTagSet(final int rColors,
+	                                               final int gColors,
+	                                               final int bColors) {
+		final int colorsInTotal = rColors*gColors*bColors;
+		final Map<String, Integer> palette = new HashMap<>(colorsInTotal);
+
+		final int rStep = 256 / rColors; //relying on down-rounding
+		final int gStep = 256 / gColors;
+		final int bStep = 256 / bColors;
+
+		for (int rColor = 0; rColor < rColors; ++rColor)
+			for (int gColor = 0; gColor < gColors; ++gColor)
+				for (int bColor = 0; bColor < bColors; ++bColor) {
+					int color = ((rColor*rStep) << 16) + ((gColor*gStep) << 8) + bColor*bStep;
```

Comment:
This line looks a bit as it could be rewritten using one of the existing rgbToValidColor() methods. Could you check this?